src/kernel/accumulate_maxsumexp/cuda.cu

```diff
@@ -32,27 +32,36 @@ void cuda_kernel(Index nelems, const T *src, T *dst)
  * */
 {
     int i = threadIdx.x + blockIdx.x*blockDim.x;
-    constexpr T zero = 0.0;
+    using Y = typename T::repr_t;
+    using Z = typename CUDAComputeType<T>::value;
```

Comment:
we do not need type Z here. It is safe to substitute type Z into type T and remove reinterpret_cast